diff --git a/src/ecat_handler/CMakeLists.txt b/src/ecat_handler/CMakeLists.txt
index da5f21e..2d661c0 100644
--- a/src/ecat_handler/CMakeLists.txt
+++ b/src/ecat_handler/CMakeLists.txt
@@ -2,6 +2,8 @@ set(LIB_NAME ecat_handler)
 
 cmake_policy(SET CMP0079 NEW)
 
+set( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fPIC" )
+set( CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -fPIC" )
 add_library(${LIB_NAME} "")
 
 target_sources(${LIB_NAME}
