diff --git a/webserver.py b/webserver.py
index b4f8ecd..a4b6129 100644
--- a/webserver.py
+++ b/webserver.py
@@ -2452,7 +2452,7 @@ if __name__ == '__main__':
                 configure_runtime()
                 monitor.parse_st(openplc_runtime.project_file)
             
-            app.run(debug=False, host='0.0.0.0', threaded=True, port=8080)
+            app.run(debug=False, host='::', threaded=True, port=8080)
         
         except Error as e:
             print("error connecting to the database" + str(e))
