diff --git a/webserver/webserver.py b/webserver/webserver.py
index 0fafa2e..fd18593 100644
--- a/webserver/webserver.py
+++ b/webserver/webserver.py
@@ -2433,7 +2433,7 @@ if __name__ == '__main__':
                 configure_runtime()
                 monitor.parse_st(openplc_runtime.project_file)
             
-            app.run(debug=False, host='0.0.0.0', threaded=True, port=8080)
+            app.run(debug=False, host='::', threaded=True, port=8080)
         
         except Error as e:
             print("error connecting to the database" + str(e))
