diff --git a/webserver.py b/webserver.py
index 0fafa2e..58b4648 100644
--- a/webserver.py
+++ b/webserver.py
@@ -1713,6 +1713,8 @@ def hardware():
             else: return_str += "<option value='rpi'>Raspberry Pi</option>"
             if (current_driver == "rpi_old"): return_str += "<option selected='selected' value='rpi_old'>Raspberry Pi - Old Model (2011 model B)</option>"
             else: return_str += "<option value='rpi_old'>Raspberry Pi - Old Model (2011 model B)</option>"
+            if (current_driver == "rpi_raspicomm"): return_str += "<option selected='selected' value='rpi_raspicomm'>Raspberry Pi - Old Model (2011 model B) with Raspi Comm</option>"
+            else: return_str += "<option value='rpi_raspicomm'>Raspberry Pi - Old Model (2011 model B) with Raspi Comm</option>"
             if (current_driver == "simulink"): return_str += "<option selected='selected' value='simulink'>Simulink</option>"
             else: return_str += "<option value='simulink'>Simulink</option>"
             if (current_driver == "simulink_linux"): return_str += "<option selected='selected' value='simulink_linux'>Simulink with DNP3 (Linux only)</option>"
