diff --git a/scripts/change_hardware_layer.sh b/scripts/change_hardware_layer.sh
index 9e70446..4cb1b3f 100755
--- a/scripts/change_hardware_layer.sh
+++ b/scripts/change_hardware_layer.sh
@@ -73,6 +73,13 @@ elif [ "$1" == "rpi_old" ]; then
     echo rpi > ../scripts/openplc_platform
     echo rpi_old > ../scripts/openplc_driver
 
+elif [ "$1" == "rpi_raspicomm" ]; then
+    echo "Activating Raspberry Pi 2011 driver with Raspi Comm"
+    cp ./hardware_layers/raspberrypi_raspicomm.cpp ./hardware_layer.cpp
+    echo "Setting Platform"
+    echo rpi > ../scripts/openplc_platform
+    echo rpi_raspicomm > ../scripts/openplc_driver
+
 elif [ "$1" == "simulink" ]; then
     echo "Activating Simulink driver"
     cp ./hardware_layers/simulink.cpp ./hardware_layer.cpp
